contracts/governance/src/collection.rs

```diff
@@ -0,0 +1,43 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+//
+// Copyright (c) DUSK NETWORK. All rights reserved.
+
+use alloc::vec::Vec;
+use canonical::CanonError;
+use canonical_derive::Canon;
+
+#[derive(Clone, Canon, Debug)]
+pub struct Collection<K, V> {
+    data: Vec<(K, V)>,
+}
+
+impl<K: PartialEq, V: PartialEq> Collection<K, V> {
+    // Methods return a result to keep things consistent with the map methods
+    pub fn get(&self, key: &K) -> Result<Option<&V>, CanonError> {
+        Ok(self.data.iter().find(|(x, _)| x == key).map(|(_, y)| y))
+    }
+
+    pub fn insert(&mut self, key: K, value: V) -> Result<(), CanonError> {
+        if let Some(pos) = self.data.iter().position(|(k, _)| k == &key) {
+            self.data[pos] = (key, value)
+        } else {
+            self.data.push((key, value))
+        }
```

Comment:
I think the version I have is more readable, is there something specific I should change?

Reply:
Your version isn't going to work since `value` gets assigned to `*v` and then we try to push it in the data array. It get's consumed by the assignment `*v = value` that's why it doesn't compile